### PR TITLE
ELPP-897 Endpoint can be validated by the RAML schema.

### PR DESCRIPTION
**Pull request by jamiehollern**
@nlisgo @giorgiosironi @thewilkybarkid

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ services:
   - mysql
 
 before_install:
+  - curl https://puli.io/installer | php
+#  - mv puli.phar /usr/local/bin/puli
```
Comment (nlisgo) [inline .travis.yml]: Do we need this line anymore?

File: Jenkinsfile [hunk 2]
```diff
@@ -4,13 +4,10 @@ elifePipeline {
...
-    // def testArtifact = "${env.BUILD_TAG}.junit.xml"
-    builderDeployRevision 'journal-cms--ci', commit
-    // only smoke tests will run inside this step:
-    builderProjectTests 'journal-cms--ci', '/srv/journal-cms'
-    // builderTestArtifact testArtifact, 'journal-cms--ci', '/srv/journal-cms/build/phpunit.xml'
-    // elifeVerifyJunitXml testArtifact
+    lock('journal-cms--ci') {
+        builderDeployRevision 'journal-cms--ci', commit
+        builderProjectTests 'journal-cms--ci', '/srv/journal-cms', ['build/phpunit.xml']
+    }
```
Comment (nlisgo) [inline Jenkinsfile]: Has @giorgiosironi requested these changes?

File: src/modules/jcms_rest/tests/src/EndpointValidatorTest.php [hunk 3]
```diff
@@ -35,30 +37,29 @@ public function setUp() {
         require_once($this->projectRoot . '/.puli/GeneratedPuliFactory.php');
       }
     }
+    // Setup Puli.
+    $this->resourceRepository = (new GeneratedPuliFactory)->createRepository();
+    /*$factoryClass = PULI_FACTORY_CLASS;
+    $this->resourceRepository = (new $factoryClass())->createRepository();*/
```
Comment (nlisgo) [inline src/modules/jcms_rest/tests/src/EndpointValidatorTest.php]: Why do we still need this code?

File: .travis.yml (hunk 1 above)
Comment (jamiehollern) [inline .travis.yml]: No, I'll remove it.

File: src/modules/jcms_rest/tests/src/EndpointValidatorTest.php (hunk 3 above)
Comment (jamiehollern) [inline src/modules/jcms_rest/tests/src/EndpointValidatorTest.php]: We don't need it at the moment but it was left in so we could use it later if the situation changes as the current implementation is hacky. Would you like it removed?

Comment (nlisgo) [inline src/modules/jcms_rest/tests/src/EndpointValidatorTest.php]: Just remove it. Would need more information to be valuable.

**Pull request closed without merging**